be/src/vec/functions/function_case.h

```diff
@@ -167,9 +167,9 @@ class FunctionCase : public IFunction {
         int rows_count = column_holder.rows_count;
 
         // `then` data index corresponding to each row of results, 0 represents `else`.
-        int then_idx[rows_count];
-        int* __restrict then_idx_ptr = then_idx;
-        memset(then_idx_ptr, 0, sizeof(then_idx));
+        std::unique_ptr<int> then_idx(new int[rows_count]);
```

Comment:
the rows_count maybe too large? I think it should be less than 4096, do you really meet this problem?

Reply:
I have meet this problem in 1.1,  the rows_count is  > 5000w, cause all backends core dump, the problem was fixed by #17035, but it is hard to pick it to branch 1.1. I think changing it on the master can also avoid the same problem again.

Reply:
@yiguolei plz help me cc